### PR TITLE
fix(config): decouple tactical room logic from strategic literals

### LINKED ISSUE #1110: P0: decouple tactical room logic from strategic room literals
## 问题描述

Owner follow-up after #1102/#1107: centralizing the active room literal is not enough. Online/runtime Screeps code should avoid depending on concrete room IDs in tactical/business logic. After a new spawn/room is selected, core creep/economy/territory behavior should work via a runtime `current_room` concept (derived from live owned rooms/spawns or explicit Memory config), not by importing the strategic focus room from a config file.

## Background / links

- Related closed issue: #1102
- Superseded PR with useful work: #1105
- Merged partial config retarget: #1107
- Owner addendum: strategic config can keep candidate/good-room lists and high-level focus, but low-level creep tactics must be decoupled from those strategic literals.

## Acceptance criteria

- Separate strategic room-selection config from runtime-current-room helpers.
- Tactical production modules do not import concrete active-room literals/config to decide creep/economy/territory behavior.
- Runtime current/owned rooms are derived from `Game.rooms`, `Game.spawns`, or explicit `Memory.runtime` config.
- Current-room adjacent scout-only targets are generated from runtime current room when explicitly enabled/synced, not hardcoded to a room ID in business logic.
- Regression/audit tests prevent protected room literals from appearing in production business modules outside the config boundary.
- `npm run typecheck`, `npm test -- --runInBand`, `npm run build`, and `git diff --check` pass.

### PR DESCRIPTION
**Pull request by lanyusea**
## Summary

- Replace the rejected PR #940-style tactical room hardcoding path with a two-layer model:
  - strategic room lists live in `prod/src/config/roomSelection.ts` / compatibility accessors;
  - tactical online code resolves the active/current room from runtime game state via `runtimeRooms` instead of scattering a concrete room literal.
- Reconcile the closed #1105 room-config branch with current `main` (`#1107`) and keep `E29N55` only as strategic config data, not as repeated business-logic constants.
- Add tests/helpers proving runtime current-room selection and strategic scout target behavior stay decoupled.

Closes #1110.
Refs #940, #1102, #1105, #1107.

## Verification

- `cd prod && npm run typecheck` ✅
- `cd prod && npm test -- --runInBand` ✅ — 98 suites / 1870 tests
- `cd prod && npm run build` ✅
- `git diff --check` ✅
- Hardcoded-room scan in `prod/src/**/*.ts`: quoted room literals appear only in `prod/src/config/roomSelection.ts` (the strategic config list); no tactical/business modules contain quoted room-name literals.

## Notes

This PR intentionally keeps the room candidate list as high-level strategy configuration only. Bottom-level creeps/task/economy logic should continue to consume runtime room context and should not use that list to decide tactical creep behavior directly.

### CODE REVIEW

File: prod/src/economy/localEnergyStrategy.ts [hunk 1]
```diff
@@ -63,7 +59,7 @@ const DEFAULT_ROOM_CONFIGS: Record<string, Omit<LocalEnergyRoomConfig, 'enabled'
     roomName,
     {
       importThreshold: DEFAULT_LOCAL_ENERGY_IMPORT_THRESHOLD,
-      sourceRooms: [DEFAULT_LOCAL_FIRST_SOURCE_ROOM],
+      sourceRooms: DEFAULT_LOCAL_FIRST_SOURCE_ROOM ? [DEFAULT_LOCAL_FIRST_SOURCE_ROOM] : [],
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/economy/localEnergyStrategy.ts]: Fixed in `6481020`: the default local-first source room now falls back to the current `roomName` when no global source-room default exists. Follow-up `8246c82` also requires `enabled: true` for memory-only configs so stale/partial Memory cannot activate imports. Verified with `npm run typecheck`, full Jest (98 suites / 1877 tests), `npm run build`, `git diff --check`, and room-literal scan.

File: prod/src/territory/expansionConfig.ts [hunk 2]
```diff
@@ -10,11 +10,198 @@ export interface TerritoryExpansionScoutTargetConfig {
...
+
+  return [
+    formatRoomName(parsed.horizontalDirection, parsed.horizontalCoordinate, parsed.verticalDirection, parsed.verticalCoordinate - 1),
+    formatRoomName(
+      parsed.horizontalDirection,
+      parsed.horizontalCoordinate + getEastwardCoordinateOffset(parsed.horizontalDirection),
+      parsed.verticalDirection,
+      parsed.verticalCoordinate
+    )
+  ].filter(isNonEmptyString);
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/territory/expansionConfig.ts:54]: Fixed in `6481020`: current-room scout-only adjacency now uses signed Screeps coordinates and returns all four cardinal neighbors (N/S/W/E). Regression coverage includes `W3N9 -> W3N10/W3N8/W4N9/W2N9` and an `E0S0` quadrant-edge case. Verified with `npm run typecheck`, full Jest (98 suites / 1877 tests), `npm run build`, `git diff --check`, and room-literal scan.

File: prod/src/config/runtimeRooms.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  const memory = (globalThis as { Memory?: Partial<Memory> }).Memory;
+  if (!memory) {
+    return;
+  }
+
+  const configuredOwnedRoomNames = getConfiguredRuntimeOwnedRoomNames();
+  const liveOwnedRoomNames = getLiveOwnedRoomNames();
+  const ownedRoomNames = getSortedUniqueRoomNames([
+    ...configuredOwnedRoomNames,
+    ...liveOwnedRoomNames
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/config/runtimeRooms.ts]: Fixed in `8246c82`: runtime room discovery now prefers live owned rooms whenever `Game.rooms`/`Game.spawns` expose ownership, only falling back to `Memory.runtime.ownedRoomNames` when no live owned rooms are observable. `refreshRuntimeRoomMemory()` overwrites stale runtime room memory with the live set and moves `currentRoomName` away from stale entries. Verified with full local gate and regression tests.

File: prod/src/config/runtimeRooms.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/config/runtimeRooms.ts]: Fixed in `8246c82`: runtime room discovery now prefers live owned rooms whenever `Game.rooms`/`Game.spawns` expose ownership, only falling back to `Memory.runtime.ownedRoomNames` when no live owned rooms are observable. `refreshRuntimeRoomMemory()` overwrites stale runtime room memory with the live set and moves `currentRoomName` away from stale entries. Verified with full local gate and regression tests.
